### DISCUSSION TITLE
Manual Mesh Concern

### DISCUSSION THREAD

**Discussion by vernonbc** [Jyers/Marlin#1031] [General Discussion]
Ender 3 V2 of course, New to 3d printing since 6 months now and stilll playing around and experimenting. I finally got around to experimenting with Mesh Leveling. I am using the 3 x 3 grid until I get the hang of it all. I finally probed the bed and created a mesh, inserted the M420 code into my PrusaSlicer and sliced a simple print. 
I have noticed that after printing the 2 prime lines that there is a pause before the actual model starts to print. The first line prints then the second line prints, nozzle raises a bit then a few seconds pause before the head moves to the model print position.
Is this considered normal? I have the M420 as my second line just after G28 in my start code. If I remove the M420 the print starts immediately after the prime lines.

Thank you

**Comment by Cal-LF** [comment]
Just reading the mesh is my thought.

**Reply by vernonbc** [reply]
Seems like an aweful long pause for a 3 x 3 mesh. I counted approx 6 seconds till it started to print my test model and I had to remove a blob of PETG from the nozzle while it oozed a bit. If it is in fact loading the mesh then you would think that the mesh would be loaded before the prime line is printed. Guess I will see what others have to say about this.
Thanks for the quick reply.

**Reply by Cal-LF** [reply]
@vernonbc Try resetting the eprom to clear any past settings. Then save ant new settings.

**Comment by Cal-LF** [comment]
I use 4X4 manual mesh and have only a slightly noticeable delay for the nozzle to move to the print start point. If I don't use the mesh, there is no delay.
I have this in start code. M420 S1; Restore Manual Mesh

**Reply by vernonbc** [reply]
Thanks for the tip, will connect to Pronterface and see what happens.

**Comment by Jyers** [comment]
Definitely interesting, I have never noticed a delay, but I have also never actually printed with manualmesh... I'll be interested if others share this experience.

**Reply by vernonbc** [reply]
Thanks, I will review my startcode and see if there is anything in there that may contribute to the pause, otherwise full reset and start over I guess.

**Reply by Jyers** [reply]
Oooo, just had a thought.

An M500 in your start gcode could contribute as it might try saving the current mesh which could take a second.

**Reply by vernonbc** [reply]
No there is no M500 in my start gcode. Maybe I should just post my start gcode. What a novel idea :)

G90 ; absolute positioning
M83 ; E relative
G28 ; home all axis
M420 S1 : mesh leveling activation
G1 Z2.0 F3000 ; raise z to prevent bed scratch
G1 X0.1 Y20 Z0.3 F5000.0 ; Move to start position
G1 X0.1 Y215.0 Z0.3 F1500.0 E15 ; Draw the first line
G1 X0.4 Y200.0 Z0.3 F5000.0 ; Move to side a little
G1 X0.4 Y20 Z0.3 F1500.0 E30 ; Draw the second line
G1 Z4.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 X5 Y20 Z4.0 F5000.0 ; Move over to prevent blob squish
G92 E0

**Reply by Jyers** [reply]
That all looks correct. Do you perhaps have powerloss recovery enabled? This can sometimes cause stange pausing issues when the SD card is a bit slow.

**Reply by Jyers** [reply]
It is also linked to Z raises, which shouldn't be triggered by mesh bed leveling compensation, but I could see it happening.

**Reply by vernonbc** [reply]
Tried disabling powerloss recover and still no luck. HOWEVER I think I may have it solved.

For the ender 3, Prusaslicer puts the following into the start gcode, I removed it for some reason and never had any issues until I started using Mesh Leveling.

M104 S[first_layer_temperature] ; set extruder temp
M140 S[first_layer_bed_temperature] ; set bed temp
M190 S[first_layer_bed_temperature] ; wait for bed temp
M109 S[first_layer_temperature] ; wait for extruder temp

I inserted this after M83 and before G28 and the problem seems to be solved so far.
I started and cancelled 3 prints and so far so good.

Thanks for all your help in this thread.

**Comment by vernonbc** [comment]
I think I may have found the issue. When I comment out the M420 it will print the 2 prime lines then immediately proceed to print the model. When I activate M420 it will print the 2 prime lines then pause because it is waiting for the extruder to meet the set temperature. The message on the display is "e heating". I am assuming this the extruder heating up (hot end heating). Seems to loose a degree or 2 when printing the prime lines.
I may try avoiding the prime lines and just go strait to printing the model with an extra few lines of skirt instead and see if that solves the pausing issue.

**Comment by vernonbc** [comment]
Played around a bit more. Removed the prime lines and added a few more lines to the skirt. M420 enabled still waits for the hotend to get to temp a second time after selecting print job even if it is +/- 1 degree. When I disabled M420 in start code, it goes directly to print right after prime lines printed.

**Comment by smooreace** [comment]
Why is there no temperature setting? Youre waiting for each to heat after
the other?

**Comment by vernonbc** [comment]
Problem solved further up the page. Last response to Jyers comment.

Thanks for all the help.
